File: ws/bonifaz/u2/WS23UnitTest/ComplexOperation/src/main/java/ec/edu/espe/complexcotroller/controller/Operation.java

package ec.edu.espe.complexcotroller.controller;

import java.math.BigDecimal;
import java.math.RoundingMode;

/**
 *
 * @author Christian Bonifaz, Code Creators, DCCO-ESPE
 */
public class Operation {


    public static float addTwoNumbers(float addend1, float addend2) {
        float sum = addend1 + addend2;
        BigDecimal bd = new BigDecimal(Float.toString(sum));
        bd = bd.setScale(2, RoundingMode.HALF_UP);
        return bd.floatValue();
    }
    
}

File: hw/bonifaz/u2/HW17TestCaseSolution/ComplexOperationsJS_corretion/ec.edu.espe.complexoperations.controler/Operation.js
module.exports = {
    addTwoNumbers: function (addend1, addend2) {
        let sum = addend1 + addend2;
        // Round to 2 decimal places
        sum = Math.round((sum + Number.EPSILON) * 100) / 100;
        return sum;
    }
};
